### PR TITLE
BL for user friends

### PR DESCRIPTION
**Pull request by sofiiamatsyhin**
Implementing business logic for user friends.
Add message, when user doesn't have friends.
Result: 

Screen Shot 2020-08-13 at 18 02 55

Screen Shot 2020-08-13 at 17 10 33

### CODE REVIEW

File: src/app/service/interceptors/interceptor.service.ts [hunk 1]
```diff
@@ -57,9 +57,10 @@ export class InterceptorService implements HttpInterceptor {
         if (error.status === FORBIDDEN) {
           return this.handle403Error(req);
         }
-        if (error.status === NOT_FOUND) {
-          return this.handle404Error(req);
-        }
+        // if (error.status === NOT_FOUND) {
+        //   // return this.handle404Error(req);
+        //   return req;
+        // }
```
Comment (Gusvarov) [inline src/app/service/interceptors/interceptor.service.ts:63]: Remove commented code

File: src/app/service/interceptors/interceptor.service.ts [hunk 2]
```diff
@@ -150,9 +151,9 @@ export class InterceptorService implements HttpInterceptor {
...
-  private handle404Error(req: HttpRequest<any>): Observable<HttpEvent<any>> {
-    console.log(`Page does not exist ${req.url}`);
-    this.router.navigate(['/error.component.html']).then(r => r);
-    return of<HttpEvent<any>>();
-  }
+  // private handle404Error(req: HttpRequest<any>): Observable<HttpEvent<any>> {
+  //   console.log(`Page does not exist ${req.url}`);
+  //   this.router.navigate(['/error.component.html']).then(r => r);
+  //   return of<HttpEvent<any>>();
+  // }
```
Comment (Gusvarov) [inline src/app/service/interceptors/interceptor.service.ts:158]: Same as lines 60-63

File: src/app/service/interceptors/interceptor.service.ts (hunk 1 above)
Comment (sofiiamatsyhin) [reply src/app/service/interceptors/interceptor.service.ts]: fixed

File: src/app/service/interceptors/interceptor.service.ts (hunk 2 above)
Comment (sofiiamatsyhin) [reply src/app/service/interceptors/interceptor.service.ts]: fixed

File: src/app/component/user/components/profile/achievements-friends/achievements-friends.component.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { PROFILE_IMAGES } from '../../../../../../assets/img/profile/profile-images';
```
Comment (mplesha) [inline src/app/component/user/components/profile/achievements-friends/achievements-friends.component.ts:2]: use alias for those paths with many `../../../`

**Comment by mplesha** [comment]
There is `console.log` in the `profile-service`
```js
console.log(`You don't have authorities to access ${req.url}`);
```
I thing it's not your, but delete it. For UI application it's OK by debugging but it don't make sense in production (with few exceptions) - who will open console in a browser to read this?

File: src/app/component/user/components/profile/achievements-friends/achievements-friends.component.ts (hunk 3 above)
Comment (sofiiamatsyhin) [reply src/app/component/user/components/profile/achievements-friends/achievements-friends.component.ts]: Fixed

**Pull request merged**